ddtrace/contrib/gunicorn/__init__.py

```diff
@@ -1,33 +1,10 @@
 """
-**Note:** ``ddtrace-run`` and Python 2 are both not supported with `Gunicorn <https://gunicorn.org>`__.
+ddtrace works with Gunicorn.
 
-``ddtrace`` only supports Gunicorn's ``gevent`` worker type when configured as follows:
-
-- The application is running under a Python version >=3.6 and <=3.10
-- `ddtrace-run` is not used
-- The `DD_GEVENT_PATCH_ALL=1` environment variable is set
-- Gunicorn's ```post_fork`` <https://docs.gunicorn.org/en/stable/settings.html#post-fork>`__ hook does not import from
-  ``ddtrace``
-- ``import ddtrace.bootstrap.sitecustomize`` is called either in the application's main process or in the
-  ```post_worker_init`` <https://docs.gunicorn.org/en/stable/settings.html#post-worker-init>`__ hook.
-
-.. code-block:: python
-
-  # gunicorn.conf.py
-  def post_fork(server, worker):
-      # don't touch ddtrace here
-      pass
-
-  def post_worker_init(worker):
-      import ddtrace.bootstrap.sitecustomize
-
-  workers = 4
-  worker_class = "gevent"
-  bind = "8080"
-
-.. code-block:: bash
-
-  DD_GEVENT_PATCH_ALL=1 gunicorn --config gunicorn.conf.py path.to.my:app
+.. note::
+    If your Gunicorn server does not run inside of ``ddtrace-run``,
+    be sure to ``import ddtrace.auto`` as early as possible in your application's
+    lifecycle.
```

Comment:
Given that gunicorn + gevent worker will patch gevent, is there a place where we can be sure we get imported before gevent? 🤔 (am I understanding this correctly?)

Or do we need guidance of "... if using a gevent worker.... please use ddtrace-run" type thing?

Reply:
This guidance reflects the setup that's proven to work by the [tests](https://github.com/DataDog/dd-trace-py/blob/3053a9417435abe8e30654237b682ae382598bba/tests/contrib/gunicorn/wsgi_mw_app.py#L16). That gives me some confidence that I don't need to be certain of my understanding of gunicorn internals, which is relevant because my answer to "am I understanding this correctly" is "I don't know".

Reply:
ah, ok. I think that test answer my question then. that everything still works for that test when using a gevent worker, then 👍🏻 

I thought I remembered us digging into the gunicorn code and it was explicitly calling gevent patch_all for the user at a certain point in the app startup/forking lifecycle. so I was worried that there was an ordering/timing of things that could still be messed up.

But if we have the tests, great!

Reply:
To my understanding, as long as we perform our cleanup (module cloning) logic before `gevent.monkey.patch_all()` is called, this should mitigate the problems we've been seeing with gevent patched modules interacting with non-gevent patched modules. What I'm not sure about is if we require `ddtrace` to be imported before `gevent`. 

If @emmettbutler or @P403n1x87 aren't sure as well, then I'd be in favour of guiding users to please run `ddtrace-run`. WDYT?